### PR TITLE
improve publish method

### PR DESCRIPTION
**Pull request by xiewenlong1202**
…video when stream has both

### CODE REVIEW

File: src/sdk/conference/channel.js [hunk 1]
```diff
@@ -77,20 +77,20 @@ export class ConferencePeerConnectionChannel extends EventDispatcher {
...
-      options.audio = !!stream.mediaStream.getAudioTracks();
+      options.audio = !!stream.mediaStream.getAudioTracks().length;
     }
     if (options.video === undefined) {
-      options.video = !!stream.mediaStream.getVideoTracks();
+      options.video = !!stream.mediaStream.getVideoTracks().length;
     }
-    if (!!options.audio === !stream.mediaStream.getAudioTracks().length || !!
-    options.video === !stream.mediaStream.getVideoTracks().length) {
+    if (!!options.audio > !!stream.mediaStream.getAudioTracks().length || !!
```
Comment (jianjunz) [inline src/sdk/conference/channel.js]: Comparing two boolean values by > looks a little bit wired.

Comment (xiewenlong1202) [reply src/sdk/conference/channel.js]: I also think so, but if not use this, the condition for if statement will be:
if ( (!!options.audio == true && !!stream.mediaStream.getAudioTracks().length == false) || 
(!!options.video == true && !!stream.mediaStream.getVideoTracks().length == false) )
I think it is much too long and not so well.

Comment (xiewenlong1202) [reply src/sdk/conference/channel.js]: add comment to this condition

File: src/sdk/conference/channel.js [hunk 2]
```diff
@@ -77,20 +77,20 @@ export class ConferencePeerConnectionChannel extends EventDispatcher {
...
-      options.audio = !!stream.mediaStream.getAudioTracks();
+      options.audio = !!stream.mediaStream.getAudioTracks().length;
     }
     if (options.video === undefined) {
-      options.video = !!stream.mediaStream.getVideoTracks();
+      options.video = !!stream.mediaStream.getVideoTracks().length;
     }
-    if (!!options.audio === !stream.mediaStream.getAudioTracks().length || !!
-    options.video === !stream.mediaStream.getVideoTracks().length) {
+    if ((!!options.audio == true && !!stream.mediaStream.getAudioTracks().length == false) ||
```
Comment (jianjunz) [inline src/sdk/conference/channel.js]: `!!options.audio == true` could be `!!options.audio`, and `!!stream.mediaStream.getAudioTracks().length == false` could be `!stream.mediaStream.getAudioTracks().length`.

**Comment by jianjunz** [comment]
Please also give a good title for this PR.

File: src/sdk/conference/channel.js (hunk 2 above)
Comment (xiewenlong1202) [reply src/sdk/conference/channel.js]: thanks for your advice, I will change that.

**Pull request merged**